packages/web-components/src/components/quote/quote.ts

```diff
@@ -0,0 +1,183 @@
+/**
+ * @license
+ *
+ * Copyright IBM Corp. 2020
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import { customElement, html, LitElement, property } from 'lit-element';
+import settings from 'carbon-components/es/globals/js/settings';
+import ddsSettings from '@carbon/ibmdotcom-utilities/es/utilities/settings/settings';
+import styles from './quote.scss';
+import StableSelectorMixin from '../../globals/mixins/stable-selector';
+import '../horizontal-rule/horizontal-rule';
+
+const { prefix } = settings;
+const { stablePrefix: ddsPrefix } = ddsSettings;
+
+export enum QUOTE_TYPES {
+  /**
+   * Default - doubleCurved
+   */
+  DEFAULT = 'doubleCurved',
+
+  /**
+   * singleCurved
+   */
+  SINGLE_CURVED = 'singleCurved',
+
+  /**
+   * singleAngle
+   */
+  SINGLE_ANGLE = 'singleAngle',
+
+  /**
+   * doubleAngle
+   */
+  DOUBLE_ANGLE = 'doubleAngle',
+
+  /**
+   * lowHighReversedDoubleCurved
+   */
+  LOW_HIGH_REVERSED_DOUBLE_CURVED = 'lowHighReversedDoubleCurved',
+}
+
+const slotExistencePropertyNames = {
+  sourceHeading: '_hasSourceHeading',
+  sourceCopy: '_hasSourceCopy',
+  sourceBottomCopy: '_hasSourceBottomCopy',
+  cta: '_hasCTA',
+};
+
+/**
+ * Quote.
+ *
+ * @element dds-quote
+ */
+@customElement(`${ddsPrefix}-quote`)
+class DDSQuote extends StableSelectorMixin(LitElement) {
+  @property({ reflect: true })
+  markType = QUOTE_TYPES.DEFAULT;
+
+  @property({ reflect: true })
+  inverse: boolean = false;
+
+  /**
+   * `true` if there is source heading.
+   */
+  protected _hasSourceHeading;
+
+  /**
+   * `true` if there is source copy.
+   */
+  protected _hasSourceCopy;
```

Comment:
Good to set the initial value, which works for type inference:

```suggestion
  protected _hasSourceCopy = false;
```